x/auth/keeper/account.go

```diff
@@ -97,7 +97,13 @@ func (k Keeper) GetAccount(ctx sdk.Ctx, addr sdk.Address) exported.Account {
 func (k Keeper) GetAllAccounts(ctx sdk.Ctx) []exported.Account {
 	var accounts []exported.Account
 	appendAccount := func(acc exported.Account) (stop bool) {
-		accounts = append(accounts, acc)
+		//not get empty coins accounts
```

Comment:
Ideally the function should preserve its functionality to retrieve all accounts, `GetAllAccounts` is intended to be used to retrieve all accounts changing this to only retrieve valid non empty accounts would change behavior in applications using this. To mitigate this you can:

Expand it to receive a filtering function 
```go
func (k Keeper) GetAllAccounts(ctx sdk.CTx, filter func([]exported.Account) []exported.Account) []exported.Account{
     	var accounts []exported.Account     
         ...
	k.IterateAccounts(ctx, appendAccount)
       return filter(accounts)
}
```

Or write a separate method with the specific filtering functionality `GetAllAccountsExport`

Reply:
Added separated method